packages/modal/src/Modal/ModalView.tsx

```diff
@@ -25,126 +25,133 @@ import {
   visibleBackdrop,
   visibleModalContentStyle,
 } from './styles';
-import { CloseIconColor, ModalProps, ModalSize } from './types';
+import { CloseIconColor, ForwardedRef, ModalProps, ModalSize } from './types';
 
 /**
  * @internal
  * Internal Modal View component
  */
-function ModalView({
-  open = false,
-  size = ModalSize.Default,
-  setOpen = () => {},
-  shouldClose = () => true,
-  darkMode: darkModeProp,
-  children,
-  className,
-  contentClassName,
-  initialFocus,
-  closeIconColor = CloseIconColor.Default,
-  ...rest
-}: ModalProps) {
-  const { theme, darkMode } = useDarkMode(darkModeProp);
+const ModalView = React.forwardRef(
+  (
+    {
+      open = false,
+      size = ModalSize.Default,
+      setOpen = () => {},
+      shouldClose = () => true,
+      closeIconColor = CloseIconColor.Default,
+      darkMode: darkModeProp,
+      id: idProp,
+      children,
+      className,
+      contentClassName,
+      initialFocus,
+      ...rest
+    }: ModalProps,
+    forwardedRef: ForwardedRef,
+  ) => {
+    const { theme, darkMode } = useDarkMode(darkModeProp);
 
-  const nodeRef = React.useRef(null);
+    const nodeRef = React.useRef<HTMLDivElement | null>(null);
+    const ref = forwardedRef ?? nodeRef;
```

Comment:
I might be wrong and it's a small nit - could we do `const nodeRef = React.useRef<HTMLDivElement | null>(forwardedRef ?? null);`